rclcpp/include/rclcpp/any_executable.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright 2014 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP_RCLCPP_ANY_EXECUTABLE_HPP_
+#define RCLCPP_RCLCPP_ANY_EXECUTABLE_HPP_
+
+#include <memory>
+
+#include <rclcpp/node.hpp>
+
+namespace rclcpp
+{
+namespace executor
+{
+struct AnyExecutable
```

Comment:
You could use the `RCLCPP_MAKE_SHARED_DEFINITIONS` macro here to add the static function `make_shared` to the `AnyExecutable` class.